denorm/formats/join.json

```diff
@@ -74,151 +97,198 @@
               "not": {
                 "properties": {
                   "joinMode": {
-                    "enum": ["async"]
+                    "enum": [
+                      "async"
+                    ]
                   }
                 },
-                "required": ["joinMode"]
+                "required": [
+                  "joinMode"
+                ]
               }
             },
             {
-              "required": ["joinKey"]
+              "required": [
+                "joinTargetKey"
+              ]
             }
           ]
         }
       ],
       "properties": {
-        "columns": {
-          "items": {
-            "$ref": "#/definitions/column"
-          },
-          "default": null,
-          "description": "Columns to select from table.",
-          "title": "Columns",
-          "type": ["null", "array"]
-        },
         "lockId": {
           "default": null,
           "description": "Advisory lock space to use for queueing. If null, it is generated from the ID and table ID.",
           "maximum": 32677,
           "minimum": -32678,
           "title": "Lock ID",
-          "type": ["integer", "null"]
+          "type": [
+            "integer",
+            "null"
+          ]
+        },
+        "tableSchema": {
+          "default": "None",
+          "description": "Name of schema.",
+          "title": "Schema",
+          "type": "string"
+        },
+        "tableName": {
+          "description": "Name of table.",
+          "title": "Table name",
+          "type": "string"
+        },
+        "tableKey": {
+          "default": null,
+          "description": "Unique key for the table.",
+          "items": {
+            "$ref": "#/definitions/keyColumn"
+          },
+          "title": "key",
+          "type": [
+            "array",
+            "null"
+          ]
+        },
+        "tableColumns": {
+          "items": {
+            "$ref": "#/definitions/column"
+          },
+          "default": null,
+          "description": "Columns to select from table.",
```

Comment:
This may be a little confusing of a description here. Maybe something like "Columns to watch for changes. If null then all columns will be watched." just to let people know this field gets used when you only want denorm to pick up changes to specific columns.

Reply:
You might include the word "source" as well, as in "source table" or "source data."